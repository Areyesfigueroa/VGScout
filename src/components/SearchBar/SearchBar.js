import React from "react";
import classes from "./SearchBar.module.css";
import SuggestionsBox from "./SuggestionsBox/SuggestionsBox";

const searchBar = (props) => {

    return (
        <div className={classes.SearchBar}>
            <input
                className={props.suggestions ? classes.collapseSearch: classes.defaultSearch}
                ref={props.searchRef}
                type="search"
                placeholder="Search..."
                id="search"
                onChange={(event) => props.search(event.target.value)} 
                onKeyUp={(event) => event.keyCode===13 ? props.select():null}/>

            <button 
            className={classes.searchButton} 
            onClick={props.select}>
                <ion-icon name="search-outline"></ion-icon>
            </button>

            <button 
            className={classes.cancelButton} 
            onClick={props.clear} 
            style={{visibility: props.showCancelBtn ? "visible":"hidden"}}>
                <ion-icon name="close-circle-outline"></ion-icon>
            </button>

            {/* If suggestions are passed then enable them. */}
            {props.suggestions ? (<SuggestionsBox data={props.suggestions}/>):null}
            {/* TESTING USE THE LINE ABOVE */}
            {/* <SuggestionsBoxContainer data={props.suggestions}/> */}
            
        </div>
    );
}

export default searchBar;