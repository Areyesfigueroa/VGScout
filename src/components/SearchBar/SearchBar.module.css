.SearchBar {
    width: 100%;
    text-align: center;
    position: relative;
    margin-top: 70px;
    box-sizing: border-box;
}

.SearchBar input[type="search"] {
    width: 100%;
    padding: 10px 33px;
    appearance: none;
    font-size: 100%;
    outline: none;
}

/* default state Transformation */
.defaultSearch {
    border: 1px solid black;
    border-radius: 20px;
}

/* SuggestionBox Transformation */
.collapseSearch {
    border: 1px solid black;
    border-radius: 20px 20px 0 0;
    border-bottom: none;
}

.SearchBar input[type="search"]::-webkit-search-cancel-button {
    display: none;
}

.SearchBar button {
    position: absolute;
    width: 33px;
    height: 100%;
    border-radius: 20px;
    background: none;
    border: 0;
    top: 0;
}

.searchButton {
    left: 0;
}

.cancelButton {
    right: 0;
}

ion-icon {
    width: 100%;
    height: 100%;
}

