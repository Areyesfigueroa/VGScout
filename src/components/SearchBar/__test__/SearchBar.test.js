import React, { createElement } from "react";

//React's UI testing library
import { render,cleanup} from '@testing-library/react';
import {toBeInTheDocument, toHaveStyle} from '@testing-library/jest-dom';

//Snapshots
import renderer from 'react-test-renderer';

//testing components
import SearchBar from '../SearchBar';

expect.extend({toBeInTheDocument, toHaveStyle});
afterEach(cleanup);

// //props.suggestions
// <SearchBar 
// searchRef={this.searchInput}
// search={this.handleSearch} 
// select={this.confirmSelection}
// clear={this.clearSearch}
// showCancelBtn={this.state.searchValue.length > 0}
// suggestions={this.state.suggestions ? this.state.suggestions:null} />

describe("Search bar component", () => {
    it("renders without crashing", () => {
        const div = createElement("div");
        const { container } = render(<SearchBar />, div);
        expect(container.firstChild.classList.contains("SearchBar")).toBeTruthy();
    });

    //Snapshot test
    it("Match Snapshot, cancelBtn and suggestions not showing", () => {
        const component = renderer.create(
            <SearchBar showCancelBtn={false} suggestions={null}/>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    //Snapshot test
    it("Match Snapshot, cancelBtn and suggestions not false/empty", () => {
        const component = renderer.create(
            <SearchBar showCancelBtn={true} suggestions={true}/>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});