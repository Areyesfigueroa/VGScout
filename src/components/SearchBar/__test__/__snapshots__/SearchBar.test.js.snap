// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component Match Snapshot, cancelBtn and suggestions not false/empty 1`] = `
<div
  className="SearchBar"
  data-testid="searchBar"
>
  <input
    className="collapseSearch"
    id="search"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
  />
  <button
    className="searchButton"
  >
    <ion-icon
      name="search-outline"
    />
  </button>
  <button
    className="cancelButton"
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    <ion-icon
      name="close-circle-outline"
    />
  </button>
</div>
`;

exports[`Search bar component Match Snapshot, cancelBtn and suggestions not showing 1`] = `
<div
  className="SearchBar"
  data-testid="searchBar"
>
  <input
    className="defaultSearch"
    id="search"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
  />
  <button
    className="searchButton"
  >
    <ion-icon
      name="search-outline"
    />
  </button>
  <button
    className="cancelButton"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    <ion-icon
      name="close-circle-outline"
    />
  </button>
</div>
`;
